src/DotLiquid/StandardFilters.cs

```diff
@@ -1048,22 +1014,47 @@ public static IEnumerable Where(IEnumerable input, string propertyName, object t
         /// <param name="targetValue">target property value</param>
         private static bool HasMatchingProperty(this object any, string propertyName, object targetValue)
         {
-            // Check if the 'any' object has a propertyName
-            object propertyValue = null;
-            if (any is IDictionary dictionary && dictionary.Contains(key: propertyName))
-            {
-                propertyValue = dictionary[propertyName];
-            }
-            else if (any != null && any.RespondTo(propertyName))
-            {
-                propertyValue = any.Send(propertyName);
-            }
-
+            var propertyValue = ResolveObjectPropertyValue(any, propertyName);
             return targetValue == null || propertyValue == null
                 ? propertyValue.IsTruthy()
                 : propertyValue.SafeTypeInsensitiveEqual(targetValue);
         }
 
+        private static object ResolveObjectPropertyValue(this object obj, string propertyName)
+        {
+            if (obj == null)
+                return null;
+            if (obj is IDictionary dictionary && dictionary.Contains(key: propertyName))
+                return dictionary[propertyName];
+            if (obj is IDictionary<string, object> dictionaryObject && dictionaryObject.ContainsKey(propertyName))
+                return dictionaryObject[propertyName];
+            var indexable = obj as IIndexable;
+            if (indexable == null)
+            {
+                var type = obj.GetType();
+                var safeTypeTransformer = Template.GetSafeTypeTransformer(type);
+                if (safeTypeTransformer != null)
+                    indexable = safeTypeTransformer(obj) as DropBase;
+                else
+                {
+                    var liquidTypeAttribute = type
+                        .GetTypeInfo()
+                        .GetCustomAttributes(attributeType: typeof(LiquidTypeAttribute), inherit: false)
+                        .FirstOrDefault() as LiquidTypeAttribute;
+                    if (liquidTypeAttribute != null)
+                    {
+                        indexable = new DropProxy(obj, liquidTypeAttribute.AllowedMembers);
+                    }
+                    else if (TypeUtility.IsAnonymousType(type) && obj.GetType().GetRuntimeProperty(propertyName) != null)
+                    {
+                        return type.GetRuntimeProperty(propertyName).GetValue(obj, null);

```

Comment:
It definitely looks more efficient to not use RespondTo, but in the case the property is undefined, it returns null. This is different than before. Is there a reason for this change?
```
return element.RespondTo(property) ? element.Send(property) : element;
```

Reply:
For this task to continue, we will need to change this behavior so it doesn't cause a breaking change, or a rationale and get @daviburg to agree to it. Without running this, it seems like a bug fix, but need developer to confirm that.

Reply:
In case this helps the case to prove it's a bug. https://replit.com/@microalps/LiquidWhereTests